web/app/view/ArrayListFilter.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2017 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+Ext.define('Traccar.view.ArrayListFilter', {
+    extend: 'Ext.grid.filters.filter.List',
+    alias: 'grid.filter.arraylist',
+
+    type: 'arraylist',
+
+    constructor: function (config) {
+        var me = this;
+        me.callParent([config]);
+        me.filter.setFilterFn(function (item) {
```

Comment:
Lets look on screenshot on device "Test 2"

"The Beginning" id = 1
"Soho" id = 5
"Broadway" id = 6

property = "Test 2".geofenceIds = [5, 6] - device in "Soho" and "Broadway"
value = filter.value = [1, 5]  - user selected "The Beginning" and "Soho"

If we used just "List" filter undercover check was array [5, 6] in array [1, 5] it is never true.
We need to check every item from first array in second.